src/content_cao.cpp

```diff
@@ -865,8 +865,9 @@ void GenericCAO::addToScene(scene::ISceneManager *smgr, ITextureSource *tsrc)
 			infostream << "serialized form: " << m_prop.wield_item << std::endl;
 			item.deSerialize(m_prop.wield_item, m_client->idef());
 		}
-		m_wield_meshnode = new WieldMeshSceneNode(smgr->getRootSceneNode(),
-			smgr, -1);
+		m_wield_meshnode = new WieldMeshSceneNode(
+			RenderingEngine::get_scene_manager()->getRootSceneNode(),
+			RenderingEngine::get_scene_manager(), -1);
```

Comment:
yes i think i will do it later, it was a first analysis about scenemgr but no time atm to continue this PR, waiting for merge to continue